doc/nrf/protocols/matter/overview/group_communication.rst

```diff
@@ -0,0 +1,265 @@
+.. _ug_matter_group_communication:
+
+Matter Group Communication
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This section describes how Group Communication works in Matter.
+
+Group Communication in the Matter standard refers to the ability of devices within a Matter network to communicate with multiple devices simultaneously using multicast messages.
+This is particularly useful for scenarios where a command needs to be sent to multiple devices at once, such as turning off all lights in a house or setting a scene.
+
+Group setup
+***********
+
+Devices are configured to be part of one or more groups.
+Each group is identified by a unique group ID.
+A device can be a member of multiple groups, and a group can include multiple devices.
+
+Group Messaging
+***************
+
+When a command needs to be sent to all devices in a group, the command initiator (a controller, a smartphone app, or another device) sends a multicast message addressed to the group ID.
+Only the devices that are members of that group will receive messages to or act on a message.
+Using Group Communication reduces network traffic because a single multicast message can reach multiple devices.
+This is more efficient than sending individual messages to each device.
+It also enhances scalability as adding more devices to a group does not increase the number of messages required for group-wide commands.
+
+.. note::
+
+   The communication reliability depends on the underlying networking technology.
```

Comment:
Worth to mention this part which Kamil pointed:

> The group communication is by design not reliable, because there is no acks or responses, so we don't know if we reached all targets or not.

Reply:
```suggestion
   The group communication is by design not reliable, because it does not use any way of acknowledgement.
   Because of that the device that sends the message is not able to determine, whether it reached all devices belonging to the group. 
   The communication reliability depends on the underlying networking technology.
```